mindsdb/integrations/handlers/sharepoint_handler/README.md

```diff
@@ -0,0 +1,100 @@
+# Sharepoint Handler
+
+Sharepoint handler for MindsDB provides interfaces to connect to Sharepoint via graph APIs and pull data into MindsDB.
+
+---
+
+## Table of Contents
+
+- [Sharepoint Handler](#Sharepoint-handler)
+  - [Table of Contents](#table-of-contents)
+  - [About Sharepoint](#about-sharepoint)
+  - [Sharepoint Handler Implementation](#sharepoint-handler-implementation)
+  - [Sharepoint Handler Initialization](#sharepoint-handler-initialization)
+  - [Implemented Features](#implemented-features)
+  - [Some useful definitions](#some-useful-definitions)
+  - [TODO](#todo)
+  - [Example Usage](#example-usage)
+
+---
+
+## About Sharepoint
+
+SharePoint in Microsoft 365 empowers teamwork with dynamic and productive team sites for every project team, department, and division. Share files, data, news, and resources. Customize your site to streamline your team’s work.
+
+## Sharepoint Handler Implementation
+
+This handler was implemented using the [Microsoft Graph API](https://learn.microsoft.com/en-us/graph/use-the-api) endpoint. 
+
+Graph API is a REST API endpoint that provides a simple and easy-to-use interface to access many microsoft tools including Sharepoint.
+
+## Sharepoint Handler Initialization
+
+The Sharepoint handler is initialized with the following parameters:
+
+- `clientId`: (required) Microsoft App client ID
+- `clientSecret`: (required) client secret associated with the App
+- `tenantId`: (required) GUID of the tenant in which the App has been created
+
+## How to get your credentials.
+
+1. Visit Microsoft Entra admin center and register a new App
+2. Go to API permissions and grant all the permissions related to sharepoint sites and resources
+3. Go to Certificates & secrets tab of the app and create a new client secret
+4. Now go to the App overview page where you will find client-ID and tenant-ID of your App.
+
+## Implemented Features
+
+- Fetch sites associated with the account and ability to update the metadata associated with a site (deletion and creation of sites has not been implemented)
+- Fetch lists associated with the account and ability to create more lists, update fields associated with lists and deletion of lists
+- Fetch site columns associated with the account and ability to create more site columns, update fields associated with site columns and deletion of site columns
+- Fetch list items associated with the account and ability to create more list items, update fields associated with list items and deletion of list items
+
+
+## Some useful definitions
+
+### Sites:
+SharePoint-sites are essentially containers for information. The way you store and organize things in SharePoint is by Sites.
+
+### Lists:
+A list is a collection of data that you can share with your team members and people who you've provided access to. You'll find a number of ready-to-use list templates to provide a good starting point for organizing list items.
+
```

Comment:
Hey @abhijitpal1247,
Lists and ListItems are clear to me, but the I don't understand what SiteColumns are. Can you give me an example?
Also, is your description of Site Columns correct here? You seem to be talking about lists here, rather than sites.

Reply:
In short site-columns define schema which is added to lists. The term "site" is added that their scope of availability is across all lists in the site. If my understanding regarding this is correct.
Adding some more links to back it up:

- https://www.tutorialspoint.com/sharepoint/sharepoint_site_column_and_content_types.htm#:~:text=Site%20Columns%20define%20reusable%20column,handlers%20to%20a%20Content%20Type.
- https://support.microsoft.com/en-au/office/create-a-column-in-a-list-or-library-2b0361ae-1bd3-41a3-8329-269e5f81cfa2

- https://learn.microsoft.com/en-us/microsoft-365/community/what-is-site-column

Please let me know if my understanding is incorrect or I my description is wrong in the README.

Reply:
@abhijitpal1247 That makes sense.
Can you do me a favor? For each description, you have added here; Sites, Site Columns etc., can you copy the official description available in the Microsoft documentation and also add the associated link?

For example, you can add this for Site Columns,
```
A Site Column is a template of a configured column. By creating a Site Column, you can reuse it anywhere else in the site and not have to manually rebuild its configuration at each reuse.

When creating a new column in a list or library, you have a choice to either "Create column" or "Add from existing site columns". Selecting the latter will add a replica of the Site Column to the location you are working.
https://learn.microsoft.com/en-us/microsoft-365/community/what-is-site-column
```

Reply:
@MinuraPunchihewa I have added definitions and links for all except "site". I could find explicit documentation on other components except that.

Reply:
Thanks, @abhijitpal1247!